public_html/json_schema_build.php

```diff
@@ -322,10 +313,67 @@ function validate_cache_id($cache_id){
         </div>
     </div>
 
-    <?php
+    <!-- Help text modal -->
+    <div class="modal fade" id="help_text_modal" tabindex="-1" role="dialog" aria-hidden="true">
+        <div class="modal-dialog modal-lg" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <span class="modal-title h4 text-monospace"></span>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                      <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <label for="help_text_input">
+                        Parameter help text is used for generating documentation and shown on demand for the command-line and web launch tools.
+                    </label>
+                    <div class="card">
+                        <div class="card-header">
+                            <ul class="nav nav-tabs card-header-tabs">
+                                <li class="nav-item"><a class="nav-link active" data-toggle="tab" href="#tab-helptext">Write</a></li>
+                                <li class="nav-item"><a class="nav-link" data-toggle="tab" href="#tab-helptext-preview">Preview</a></li>
+                            </ul>
+                        </div>
+                        <div class="card-body tab-content">
+                            <div class="tab-pane fade show active" id="tab-helptext" role="tabpanel">
+                                <div class="form-group help_text_modal_group">
+                                    <textarea class="form-control" id="help_text_input" rows="5"></textarea>
+                                    <small class="form-text text-muted">
+                                        You can use <a href="https://www.markdownguide.org/cheat-sheet/" target="_blank">markdown</a>,
+                                        but remember that this will be shown raw on the command line. So no tables please!
+                                    </small>
+                                </div>
+                            </div>
+                            <div class="tab-pane fade" id="tab-helptext-preview" role="tabpanel">
+                                <p>Command-line:</p>
+                                <pre><span class="helptext-preview-title"></span>
+<span class="helptext-preview-description"></span>
+
+<span class="helptext-preview-helptext text-muted"></span></pre>
```

Comment:
how about if we call this class `helptext-preview-text`, sound less confusing to me 🙂

Reply:
Yeah, could do. It makes more sense in the javascript though as the variable we put in here is called `helptext`